js/index.js

```diff
@@ -1,42 +1,84 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
+  //console.log('Calculating subtotal, yey!');
+let price = product.querySelector('.price span').innerText  
+let quantity = +product.querySelector('.quantity input').value;
   //... your code goes here
+let subtotal = price * quantity
+
+product.querySelector('.subtotal span').innerText = subtotal
+//console.log(subtotal)
+/* console.log(document.querySelector('td.price span').innerText)
+console.log(document.querySelector('.quantity input').value)
+console.log(document.querySelector('.subtotal span').innerText) */
+//these console logs only works when "calculate prices" button
+//on the webpage is clicked.
+//console.log(price, quantity)
+return subtotal
+
 }
 
+
+
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+/*   const singleProduct = document.querySelector('.product');
+  updateSubtotal(singleProduct); */
   // end of test
 
   // ITERATION 2
   //... your code goes here
-
+  let products = document.querySelectorAll('.product')
+  for(let i = 0; i<products.length; i++){
```

Comment:
You don't need the first loop, because you update the subtotal value also in the second loop